runtime/integration-tests/src/test_macros.rs

```diff
@@ -461,9 +461,97 @@ macro_rules! parachain1_transfer_asset_to_parachain2_and_back {
 	}};
 }
 
+macro_rules! transfer_native_token_from_pendulum_to_assethub {
+	(
+        $mocknet:ident,
+        $pendulum_runtime:ident,
+        $pendulum_chain:ident,
+        $assethub_runtime:ident,
+        $assethub_chain:ident,
+        $pendulum_id:ident,
+        $assethub_id:ident
+    ) => {{
+		use crate::mock::{units, BOB};
+		use frame_support::traits::fungibles::Inspect;
+		use polkadot_core_primitives::Balance;
+		use xcm::latest::{
+			Junction,
+			Junction::AccountId32,
+			Junctions::{X1, X2},
+			MultiLocation, WeightLimit,
+		};
+		use $pendulum_runtime::CurrencyId;
+
+		$mocknet::reset();
+
+		let transfer_amount: Balance = units(10);
+		let asset_location = MultiLocation::new(
+			1,
+			X2(Junction::Parachain($pendulum_id), Junction::PalletInstance(10)),
+		);
+
+		// Get BOB's balance before the transfer on Pendulum chain
+		let mut pendulum_tokens_before: Balance = units(100);
+		$pendulum_chain::execute_with(|| {
+			assert_eq!(
+				$pendulum_runtime::Tokens::balance(CurrencyId::Native, &BOB.into()),
+				pendulum_tokens_before
+			);
+		});
+
+		// Execute the transfer from Pendulum chain to AssetHub
+		$pendulum_chain::execute_with(|| {
+			use $pendulum_runtime::XTokens;
+
+			assert_ok!(XTokens::transfer_multiasset(
+				$pendulum_runtime::RuntimeOrigin::signed(BOB.into()),
+				Box::new((asset_location.clone(), transfer_amount).into()),
+				Box::new(
+					MultiLocation {
+						parents: 1,
+						interior: X1(AccountId32 { network: None, id: BOB })
+					}
+					.into()
+				),
+				WeightLimit::Unlimited
+			));
+		});
+
+		$pendulum_chain::execute_with(|| {
+			use $pendulum_runtime::{RuntimeEvent, System};
+
+			assert!(System::events().iter().any(|r| matches!(
+				r.event,
+				RuntimeEvent::XTokens(orml_xtokens::Event::TransferredMultiAssets { .. })
+			)));
+		});
+
+		// Can't get this to work though
+		// Verify the balance on the AssetHub chain after transfer
+		// $assethub_chain::execute_with(|| {
+		// 	   // I need ForeignAssets here instead of Assets
+		//     use $assethub_runtime::Assets;
+
+		//     assert_eq!(
+		//         Assets::balance(asset_location, &BOB.into()),
+		//         transfer_amount
+		//     );
+		// });
```

Comment:
Using 
```
use $assethub_runtime::ForeignAssets;
ForeignAssets::balance(asset_location, &BOB.into()),
```
doesn't work?

Reply:
No, unfortunately

Reply:
Okay nevermind, of course it does not work because the asset_hub dependency we are using is [this](https://github.com/pendulum-chain/pendulum/blob/9d5791f2b12db461b3d7594f2a9e61d7d534ac10/runtime/integration-tests/Cargo.toml#L39C30-L39C67) one.

And ForeignAssets does not exist [here](https://github.com/paritytech/cumulus/blob/e05c8d7f71734ed71188337c6cb0d30715f6320f/parachains/runtimes/assets/statemint/src/lib.rs#L635).

We would probably need to replace the `statemint` and `statemine` dependencies we are defining in the Cargo.toml with the more recent asset hubs [here](https://github.com/paritytech/polkadot-sdk/tree/master/cumulus/parachains/runtimes/assets). Can you try that @bogdanS98? I am rather pessimistic that this works though as this will probably cause compilation issues with cargo due to dependency mismatches. Still worth a shot though.